src/components/other/ProjectAddForm/index.js

```diff
@@ -58,142 +60,131 @@ const mapDispatchToProps = dispatch => ({
-                    <DangerButton onClick={this.onModalClose}>
-                        {_ts('components.addProject', 'modalCancel')}
-                    </DangerButton>
-                    <PrimaryButton
-                        type="submit"
-                        disabled={pending || !pristine}
-                    >
-                        {_ts('components.addProject', 'modalCreate')}
-                    </PrimaryButton>
-                </div>
-            </Faram>
-        );
-    }
+const projectAddSchema = {
+    fields: {
+        title: [requiredCondition],
+        isPrivate: [],
+        organizations: [],
+    },
+};
+
+function ProjectAddForm(props) {
+    const {
+        className,
+        onModalClose,
+        userGroups,
+        userId,
+        onProjectAdd,
+        setUserProject,
+        setUsergroupProject,
+        setUserProfileProject,
+    } = props;
+
+    const [faramValues, setFaramValues] = useState({ isPrivate: false, organizations: [] });
+    const [faramErrors, setFaramErrors] = useState({});
+    const [pristine, setPristine] = useState(true);
+
+    const [
+        pending,
+        ,
+        ,
+        createNewProject,
+    ] = useRequest({
+        url: 'server://projects/',
+        body: {
+            title: faramValues.title,
+            userGroups,
+            isPrivate: faramValues.isPrivate,
+            organizations: faramValues.organizations,
+        },
+        method: 'POST',
+        onSuccess: (response) => {
+            setUserProject({ project: response });
+            setUsergroupProject({ project: response });
+            setUserProfileProject({
+                userId,
+                project: response,
+            });
+            if (onProjectAdd) {
+                onProjectAdd(response.id);
+            }
+            notify.send({
+                title: _ts('components.addProject', 'userProjectCreate'),
+                type: notify.type.SUCCESS,
+                message: _ts('components.addProject', 'userProjectCreateSuccess'),
+                duration: notify.duration.MEDIUM,
+            });
+            onModalClose();
+        },
+        onFailure: (error, errorBody) => {
+            setFaramErrors(errorBody?.faramErrors);
+            setPristine(true);
+            notifyOnFailure(_ts('components.addProject', 'addProjectModalLabel'))({ error: errorBody });
+        },
+    });
+
+    const handleFaramChange = useCallback((newFaramValues, newFaramErrors) => {
+        setFaramValues(newFaramValues);
+        setFaramErrors(newFaramErrors);
+        setPristine(false);
+    }, [setFaramValues, setFaramErrors, setPristine]);
+
+    const handleFaramFailure = useCallback((newFaramErrors) => {
+        setFaramErrors(newFaramErrors);
+        setPristine(true);
+    }, [setFaramErrors, setPristine]);
+
+    const handleFaramSuccess = useCallback(() => {
+        createNewProject();
+    }, [createNewProject]);
```

Comment:
const handleFaramSuccess = createNewProject;

Reply:
if it is so then can't we use createNewProject directly as props to Faram.

Reply:
we can yes sir!